GeoLib/Plane.h

```diff
@@ -0,0 +1,118 @@
+/**
+ * \brief  Definition of the class Plane - a plane in 3d space described in
+ * Hessian normal form.
+ *
+ * \copyright
+ * Copyright (c) 2012-2015, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef OGS_PLANE_H
+#define OGS_PLANE_H
+
+#include <utility>
+
+#include "MathLib/Vector3.h"
+
+namespace GeoLib
+{
+
+class Plane final
+{
+public:
+	Plane(MathLib::Vector3 const& unit_normal, double d)
+		: _unit_normal(unit_normal), _oriented_distance(d)
+	{}
+
+	/// Constructs the hessian normal form of the plane spanned by the
+	/// vectors \f$v\f$ and \f$w\f$ attached to the position vector \f$p\f$
+	Plane(MathLib::Vector3 const& v, MathLib::Vector3 const& w,
+		MathLib::Vector3 const& p)
+		: _unit_normal(MathLib::crossProduct(v,w).getNormalizedVector()),
+			_oriented_distance(-1.0 * MathLib::scalarProduct(_unit_normal, p))
+	{}
+
+	bool isPointInPlane(MathLib::Vector3 const& p,
+		double tol=std::numeric_limits<float>::epsilon()) const
+	{
+		return getDistance(p)<tol;
+	}
+
+	/// Computes the distance between the position vector \f$p\f$ and the
+	/// plane
+	double getDistance(MathLib::Vector3 const& p) const
+	{
+		return MathLib::scalarProduct(_unit_normal,p)+_oriented_distance;
+	}
+
+	/// Returns the normalized normal of the plane.
+	MathLib::Vector3 const& getNormal() const
+	{
+		return _unit_normal;
+	}
+
+	/// Returns the distance to the origin.
+	double getDistance() const { return _oriented_distance; }
+
+private:
+	MathLib::Vector3 _unit_normal;
+	double _oriented_distance;
+};
+
+/// Function computes the intersection of two planes
+/// @param p0 first plane
+/// @param p1 second plane
+/// @return a line described by the pair of a vector (direction) and a point of
+/// the line
```

Comment:
Add prerequisite: It must be checked beforehand that the two planes actually intersect.
Or return pair(0, 0) in the case of parallel planes?